core/win32/drwinapi/kernel32_proc.c

```diff
@@ -49,7 +49,7 @@ kernel32_redir_init_proc(void)
 {
     PEB *peb = get_own_peb();
     ASSERT(get_os_version() < WINDOWS_VERSION_2003 ||
-           peb->FlsBitmap->SizeOfBitMap == FLS_MAX_COUNT);
+           (peb->FlsBitmap == NULL || peb->FlsBitmap->SizeOfBitMap == FLS_MAX_COUNT));
 #ifdef CLIENT_INTERFACE
     /* We rely on -private_peb for FLS isolation.  Otherwise we'd have to
```

Comment:
It seems like the comment below applies: your PR here fixes the immediate crash, but we still have no FLS isolation, and so we'll have mixed up FLS callbacks and ensuing problems.  Please put a comment here about this, sthg like:
```
/* FIXME i#3633: Implement FLS isolation in Win10 1903+ where FLS data is no longer
 * in the PEB.
 */
```

Reply:
I think that FLS is properly isolated atm, and it was before. FLS only comes from CRT_INIT in the client dll.

As far as I've checked (correct me if I'm wrong), there is hook of `GetProcAddress` in client dll done by `dynamorio.dll`, this one leads to `redir_LdrGetProcAddress` and redirects `Fls APIs` to `redir_Fls*` functions. Thus FLS allocations/freeing is done only for client in separated `private_peb`. 

Only problem (if it is a problem at all) that I see, is that `redir_RtlProcessFlsData` won't be called on 1903 by `LdrShutdownProcess/Thread` as it is private symbol thus it won't be intercepted, but since on `thread_exit` entry point of the client is called, it will trigger `FlsFree`, and lead to freeing allocated FLS data.

Correct me if I'm wrong, but this is what I understood by looking over code.

Reply:
OK, if we redirect everything.  I have not looked at these Windows details in a long time.  Your comment in the commit message "Temporarily fixes issue" then should perhaps be reworded?  And maybe your "Better solution..." comment should be replicated in ntdll_redir_fls_init() so it's clear to the reader that we're deliberately keeping our private FLS data in private_peb even on 1903 where the app is keeping it in ntdll globals.

Reply:
I would prefer to call it temporarily for the moment, as 1903 is still fresh, and I don't know what Microsoft will/might change in the upcoming weeks. Ideal solution would be to put `FlsBitmap`, `FlsCallback`, `FlsListHead` and `FlsHighIndex` as global variables with `redir_` prefix and completely remove its usage in PEB. Could happen that Microsoft decides to use this filed(s) in PEB for something else, which would possibly conflict with FLS usage in private_peb. 

It is your call.